src/metabase/models/params/chain_filter.clj

```diff
@@ -115,26 +115,21 @@
 
 (defn- filter-clause
   "Generate a single MBQL `:filter` clause for a Field and `value` (or multiple values, if `value` is a collection)."
-  [source-table-id field-id value]
+  [source-table-id {:keys [field-id op value options]
+                    :or   {op :=}}]
   (let [field-clause (let [this-field-table-id (field/field-id->table-id field-id)]
                        [:field field-id (when-not (= this-field-table-id source-table-id)
                                           {:join-alias (joined-table-alias this-field-table-id)})])]
-    (cond
-      ;; e.g. {$$venues.price [:between 2 3]} -> [:between $venues.price 2 3]
-      ;; this is not really supported by the API directly
-      (and (sequential? value) (keyword? (first value)))
-      (into [(first value) field-clause] (rest value))
-      ;; e.g. {$$venues.price #{2 3}} -> [:= $$venues.price 2 3]
-      (and (coll? value) (not (map? value)))
-      (into [:= field-clause] value)
-      :else
-      ;; e.g. {$$venues.price "past32weeks"} -> [:time-interval $checkins.date -32 :week]
-      (or (when (and (temporal-field? field-id)
-                     (string? value))
-            (u/ignore-exceptions
-              (params.dates/date-string->filter value field-id)))
-          ;; e.g. {$$venues.price 2} -> [:= $$venues.price 2]
-          [:= field-clause value]))))
+    (if (and (temporal-field? field-id)
+             (string? value))
+      (u/ignore-exceptions
+        (params.dates/date-string->filter value field-id))
+      (cond-> [op field-clause]
+        true          (into (if (or (sequential? value)
+                                    (set? value))
+                              value
+                              [value]))
```

Comment:
```suggestion
      [op field-clause (u/one-or-many value)]
```

this can be replaced with one of our helper funciton.

Reply:
done in 39ec5a256ca0cd926f3cbad9232029374e1191fe (though it's not as elegant as just calling that function because of nils)